lib/core/Stage.cpp

```
@@ -98,7 +98,15 @@ void Stage::start() {
         pid_t tid = syscall(SYS_gettid);
         register_tid(tid);
 #endif
-        main_thread_fn(std::ref(*this));
+        try {
+            main_thread_fn(std::ref(*this));
+        } catch (const FatalError& e) {
+            // FATAL_ERROR already called exit_kotekan; avoid terminating the thread abruptly.
+            ERROR("Stage {:s} caught FatalError: {:s}", unique_name, e.what());
+        } catch (const std::exception& e) {
+            ERROR("Exception in stage {:s}: {:s}", unique_name, e.what());
+            throw;
```

Should we throw the exception we just got?

---

`throw;` with no arguments should do just this.